operators/ior_presets.py

```diff
@@ -0,0 +1,181 @@
+    _ior_values["air_0C"] = ["Air @ 0C", 1.000293]
+    _ior_values["air_stp"] = ["Air @ STP", 1.000277]
+    _ior_values["amber"] = ["Amber", 1.55]
+    _ior_values["blood"] = ["Blood", 1.301]
+    _ior_values["calcite_e"] = ["Calcite (extraordinary)", 1.486]
+    _ior_values["calcite_o"] = ["Calcite (ordinary)", 1.658]
+    _ior_values["cellulose"] = ["Cellulose", 1.47]
+    _ior_values["crown_glass_pure"] = ["Crown glass (pure)", 1.52]
+    _ior_values["cubic_zirconia"] = ["Cubic zirconia", 2.15]
+    _ior_values["diamond"] = ["Diamond", 2.415]
+    _ior_values["ethanol_20C"] = ["Ethanol @ 20C", 1.36]
+    _ior_values["flint_glass_dense"] = ["Flint glass (dense)", 1.71]
+    _ior_values["flint_glass_pure"] = ["Flint glass (pure)", 1.61]
+    _ior_values["glycerine"] = ["Glycerine", 1.47]
+    _ior_values["nacl"] = ["Sodium Chloride", 1.544]
+    _ior_values["olive_oil"] = ["Olive Oil", 1.46]
+    _ior_values["opal"] = ["Opal", 1.44]
+    _ior_values["pet_plastic"] = ["PET plastic", 1.5750]
+    _ior_values["plate_glass"] = ["Plate Glass", 1.52]
+    _ior_values["polycarbonate"] = ["Polycarbonate", 1.584]
+    _ior_values["polystyrene"] = ["Polystyrene", 1.52]
+    _ior_values["pyrex_glass"] = ["Pyrex Glass", 1.470]
+    _ior_values["quartz"] = ["Quartz", 1.458]
+    _ior_values["sapphire"] = ["Sapphire", 1.77]
+    _ior_values["silicone_oil_20C"] = ["Silicone Oil @ 20C", 1.4]
+    _ior_values["turpentine"] = ["Turpentine", 1.47]
+    _ior_values["vacuum"] = ["Vacuum", 1]
+    _ior_values["water_20C"] = ["Water @ 20C", 1.330]
+    _ior_values["water_ice"] = ["Water Ice", 1.31]
+
+    # A default value that will be assigned an integer index of 0 as required
+    # by EnumProperty when using a callback to obtain the list of items.
+    default_key = "water_20C"
+
+    def get_sorted_list(self, sort="name"):
+        # Make a deep copy of the ior_values dict
+        ior_values = copy.deepcopy(self._ior_values)
+        # Append a unique integer index to each tuple. The EnumProperty needs
+        # this to determine a default item (index 0) when using a callback.
+        # Failure to do so results in a random item becoming the default.
+        ior_values[self.default_key].append(0)
+        index = 1
+        for ior in ior_values:
+            if ior != self.default_key:
+                ior_values[ior].append(index)
+                index += 1
+        # Convert the dict to a sorted list of tuples
+        if sort == "value":
+            index = 1
+        else:
+            index = 0
+        preset_list = []
+        for item in sorted(ior_values.items(),
+                           key=lambda e: e[1][index]):
+            text = "{} ({:f})".format(item[1][0], item[1][1])
+            preset_list.append((item[0], text, text, item[1][2]))
+        return preset_list
+
+    @classmethod
+    def get_item(cls, key=default_key):
+        ret = None
+        if key not in cls._ior_values.keys():
```

Comment:
I think this can never happen?

I would just replace the method body with
`return cls._ior_values[key]`

Similar in `get_value` below.